### ISSUE TITLE
Is there any alive fork with the app in AppStore?

### ISSUE THREAD

**Issue by SVilgelm** [MiniKeePass/MiniKeePass#621]
Seems, this project is died. Does anyone know the alive fork with the App in AppStore?

**Comment by sir106** [comment]
how about KeePass Touch? (https://itunes.apple.com/de/app/keepass-touch/id966759076?mt=8)

V1.9.1 was released about a month ago ... I switched also to KPT and didn't regret it...

**Comment by SiGamma** [comment]
Source code isn't available on GitHub and people haven't been able to get the dev to send it to them, so that's a dealbreaker for me.

**Comment by SVilgelm** [comment]
I agree with @SiGamma, I cannot trust the KPT because I don't see its code.

**Comment by mstarke** [comment]
The developer seems to respond if you're really patient. @Jellyfrog has had luck and published the code: https://github.com/Jellyfrog/KeePass-Touch

**Comment by mstarke** [comment]
And AFAIK KeePassTouch isn't compliant with the licensed source code, since GPL software isn't compatible with the AppStore. It's actually quite sad that MiniKeePass never received any funding and provided an add free App and now someone else is riding the train, violating the license (MiniKeePass is GPL) and even earns something from the adds and in app purchases.

**Comment by skyline75489** [comment]
I've been longing for the comeback of this project.  It'd be great to integrate MiniKeePass with [KeePassKit](https://github.com/MacPass/KeePassKit).

**Comment by ConcurrentHashMap** [comment]
@jamisonbennett I would like to discuss this on here... Maybe we can create an official, maintained fork and start on getting MiniKeePass on an up-to-date version support KDBX4, FaceID and iOS 12 password manager API. I know there are a lot of good programmers out there that already forked and fixed things, but we need to get those changes on the App Store. I would be open to help as best as I can (although I don't have any experience in ObjectiveC or Swift so far), but I would also be open to donate or to supply developer licenses for App Store by myself and support the development process by testing and management of tickets etc.

Anyone +1 for getting this working again?

**Comment by 4brunu** [comment]
There are a lot of PR open, but they don't seem to be accepted...
I like this project, and I would like to avoid the fork scenario, maybe if @jarush and @tallerthenyou are willing to add some people to the organisation, the project could go on.

**Comment by jamisonbennett** [comment]
@ConcurrentHashMap I have tried reaching out to the maintainers multiple times including attempting to send a "wake up call" via their PayPal donation link. @jarush and @tallerthenyou, as I said in my private direct messages, I want to avoid forking and have asked to be added to the GitHub project. My last attempt was less than a week ago, so I am going to give it a little time. However, if I do not hear anything, I am considering filing the paperwork for an LLC to limit my personal liability (but I am still undecided on this) and then I hope to revive the community using a forked repo.

We will have to change the name of the App in order to get it published on the App Store. But I want the source code the be open source so that it can be audited (and in this case it is a legal requirement based on the license). If we go the forking route, I suspect it may take a 2-3 months to get these fixes published due to LLC paperwork, US export compliance paperwork (yes, this still has to be done but its much easier for open source now), and the App store review. But for the record, I am completely against hostile forks because it leaves the community fragmented. In this case there are vulnerabilities in security software and the community seems unresponsive so I am willing to consider making an exception in this case.

As for the next features, I really like the iOS 12 password manager API. I have started to prototype that in my spare time.

**Comment by SVilgelm** [comment]
I'm in. I like how KeepassXC team works. So, it would be great to have something similar.

**Comment by ConcurrentHashMap** [comment]
@jamisonbennett Thanks for your quick reply! I‘m totally up to you and like the way you want to go and I would like support that in any way :+1: The best would be a reaction of @jarush or @tallerthanyou but I also didn‘t find a way of current contact addresses... Unmaintained repositories are really a problem on GitHub. So if this doesn‘t work we should make a hard fork and name it different. 

I‘m really looking forward to a good and fresh community and your prototype sounds promising! :-)

**Comment by mstarke** [comment]
I'm very happy if MiniKeePass gets a push and development is dragged out of this limbo and am very thankful for such high spirit in the community. But the code is licensed under GPL and thus not allowed to be published as App in the App Store. The original owners are free to do whatever they want to the code but any fork has to adhere to the GLP license and have to comply to it. I have the same dilemma with MacPass since I re-used some code of the repo. A way to circumvent this would be to request a new license from the owners (and all contributors!) but since the repo owners aren't really vocal anymore this seems to be an impossible task.

**Comment by ConcurrentHashMap** [comment]
@mstarke Thanks for pointing that out. This might be a legal issue, but I hope this might not be unsolvable. In the worst case we cannot distribute the binary inside the App Store as long as we didn't get contact with the original developers, so everyone who wants to use it, will need an Apple developer account (please correct me if I'm not up-to-date here, but as long as know this is free if you don't want to publish on the App store) and might be able to side-load it onto his phone. Pro: You can be sure the app you put on your phone is 100% equivalent to the source you downloaded. Negativ: Updates won't be easy like in the App Store. Better way though than using an 2-year old app with several security issues.

**Comment by jamisonbennett** [comment]
@mstarke Thank you for pointing out the complication that forks will have in the Apple App Store with respect to the GPL license. I was not aware of those complications, but it is pretty clear that App Store adds additional restrictions which are not in the GPL. As a result that is why many GPL apps are no longer being approved during the App Store review process. The original authors have an advantage here since they own the code and can dual license it as needed in order to comply with the App Store restrictions. We have to hope the original authors respond, otherwise the main iOS distribution channel is not a viable option for a fork. Although it is possible to load a fork on your personal device using Xcode (if you have a mac) and it is now free to do so (you do not need the paid developer account like you used to but you do need to register as a developer and create your own certificate and team but this has the downside that the app has a maximum duration of 7 days on the device since, that's all you can get with the free tier), in my opinion this method is not going to be widely adopted. This method of manually loading apps might be ok for developers, but isn't really going to have much of an impact beyond a very small group of highly technical people. The majority of the users of MiniKeePass which I know of fall into the category of uses which just wants to use the App Store.

**Comment by barrymelrose** [comment]
I'm a little late to this thread but have been loosely following. As anyone tried them on Twitter (@rush0033 or @JohnFlanagan) or Facebook? I found Flanagan on LinkedIn and sent him a message; nothing for Rush but I didn't spend a ton of time on it. I'm not on Facebook but that would be the next spot to try for these guys. I'm not a developer but I've been using this app for quite some time and would love to help facilitate getting it available on the app store again.
